piqe_ocp_lib/api/tasks/operator_ops.py

```diff
@@ -117,6 +117,32 @@ def is_operator_installed(self, operator_name: str, operator_namespace: str) ->
         """
         return self.csv.get_cluster_service_version(operator_name, operator_namespace) is not None
 
+    def get_version_of_operator(self, operator_name: str, operator_namespace: str) -> str:
+        """
+        Get the version of operator if operator is installed
+        :param operator_name: name of the operator.
+        :param operator_namespace: namespace of the operator
+        return: version of the operator
+        """
+        if self.is_operator_installed(operator_name, operator_namespace) is True:
+            return self.ohp_obj.get_package_manifest(operator_name).status.channels[0]['currentCSVDesc']['version']    
```

Comment:
Accessing the attributes here and pulling the first one off the list makes me a bit nervous. It seems that on some code paths `get_package_manifest` can return different types (None, List, object). This is one that keeps coming up, cc @btjd @Dannyb48

You can see discussion about this method at https://github.com/piqe-test-libraries/piqe-ocp-lib/pull/56#issuecomment-896214530 and https://github.com/piqe-test-libraries/piqe-ocp-lib/pull/52

Reply:
So I had a conversation with Anand on how this method should be reworked and we agree on the following algorithm:

Implement a new method for the Subscription class called get_all_subscriptions()
To check that an operator is installed:

1. Get all subscriptions and loop through them until you find the one associated with your operator
2. Retrieve spec.startingCSV and metadata.namespace from the subscription object and use these params to call is_cluster_service_version_present.

Basically the algorithm consists of finding the subscription that maps to the target operator, retrieve the csv name and operator namespace from the subscription. We can verify that an operator is successfully installed if we have successfully subscribed to it, which would result in a csv in the same namespace where the subscription and operatorgroup are created for this operator. Bonus points would be to check the status conditions of the subscription to check the subscription was successfully created.

Reply:
Thanks, @jbpratt, @btjd!